hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/BigQueryStorageHandlerBase.java

```diff
@@ -209,6 +217,45 @@ public void addExecHook(String hookType, Class<? extends ExecuteWithHookContext>
     }
   }
 
+  public Schema getBigQuerySchema(Injector injector, TableDesc tableDesc, TableId tableId) {
+    if (HCatalogUtils.isHCatalogOutputJob(conf)) {
+      // With Pig and HCatalog on Dataproc, we currently cannot the Hive Metastore to retrieve
+      // the Hive schema. So we fetch the schema directly from BigQuery, which means that the
+      // BigQuery table must already exist.
+      // TODO: See if we can remove this limitation
+      TableInfo bqTableInfo = injector.getInstance(BigQueryClient.class).getTable(tableId);
+      if (bqTableInfo == null) {
+        throw new RuntimeException("BigQuery table does not exist: " + tableId);
+      }
+      return bqTableInfo.getDefinition().getSchema();
+    }
+    // Fetch the Hive schema
```

Comment:
can we combine both into the logic of 

if (fetch_schema_from_bigquery)  else (fetch_schema_from_hive), or if we know if table exists in bq, if (exist) fetch_from_bq else fetch_from_hive?